File: backend/mahapro/src/main/java/com/mahapro/backend/mahapro/model/Transaction.java
package com.mahapro.backend.mahapro.model;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.mahapro.backend.mahapro.model.Business.Branch.Branch;
import com.mahapro.backend.mahapro.model.Business.Discount.Discount;
import com.mahapro.backend.mahapro.model.User.User;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name="transaction")
public class Transaction {

    @Id
    @GeneratedValue(strategy=GenerationType.AUTO)
    @Column(name="transaction_id")
    private int id;

    @Column(name="discount_coupon_code")
    private String discountCouponCode;

    @Column(name="used_on")
    private Double usedOn;
    
    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name="discount_id")
    private Discount discount;

    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name = "branch_id")
    private Branch branch;

    @Column(name = "status")
    private String status;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getDiscountCouponCode() {
        return discountCouponCode;
    }

    public void setDiscountCouponCode(String discountCouponCode) {
        this.discountCouponCode = discountCouponCode;
    }

    public Double getUsedOn() {
        return usedOn;
    }

    public void setUsedOn(Double usedOn) {
        this.usedOn = usedOn;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Discount getDiscount() {
        return discount;
    }

    public void setDiscount(Discount discount) {
        this.discount = discount;
    }

    public Branch getBranch() {
        return branch;
    }

    public void setBranch(Branch branch) {
        this.branch = branch;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    @Override
    public String toString() {
        try {
            return new ObjectMapper().writeValueAsString(this);
        } catch (Exception f) {
            System.out.println(f.getMessage());
            return null;
        }
    }
}

File: customer_frontend/lib/models/transaction.dart
import 'package:studio_projects/models/branch.dart';
import 'package:studio_projects/models/discounts/discount.dart';

class Transaction {
  int _transactionId;
  Discount _discount;
  String? _couponCode;
  double? _usedOn;
  Branch? _branch;
  String _status;

  Transaction.name(this._transactionId, this._discount, this._couponCode, this._usedOn, this._branch, this._status);

  String get status => _status;

  set status(String value) {
    _status = value;
  }

  Branch? get branch => _branch;

  set branch(Branch? value) {
    _branch = value;
  }

  double? get usedOn => _usedOn;

  set usedOn(double? value) {
    _usedOn = value;
  }

  String? get couponCode => _couponCode;

  set couponCode(String? value) {
    _couponCode = value;
  }

  Discount get discount => _discount;

  set discount(Discount value) {
    _discount = value;
  }

  int get transactionId => _transactionId;

  set transactionId(int value) {
    _transactionId = value;
  }

  static Transaction toClass(Map<String, dynamic> json) {
    return Transaction.name(
        json['id'],
        Discount.toClass(json['discount']),
        json['discountCouponCode'],
        json['usedOn'],
        json['branch'] != null ? Branch.toClass(json['branch']) : null,
        json['status']
    );
  }
}